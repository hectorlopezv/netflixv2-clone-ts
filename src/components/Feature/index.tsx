import React from 'react';
import {Container, Title, SubTitle } from './styles/Feature';

export interface FeatureProps {
    
}
 
const Feature: React.FC<FeatureProps> = ({children, ...props}) => {
    return (  
        <Container {...props}>{children}</Container>
    );
}
 
export default Feature;

export interface Feature_TitleProps {
    
}
 
export const Feature_Title: React.FC<Feature_TitleProps> = ({children, ...props}) => {
    return ( <Title {...props}>{children}</Title>  );
}


export interface Feature_SubTitleProps {
    
}
 
export const Feature_SubTitle: React.FC<Feature_SubTitleProps> = ({children, ...props}) => {
    return (  
        <SubTitle {...props}>{children}</SubTitle>
    );
}
 
