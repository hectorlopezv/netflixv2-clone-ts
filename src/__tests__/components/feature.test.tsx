/* eslint-disable react/jsx-pascal-case */
import React from 'react';
import { render, fireEvent, getByTestId, getByAltText } from '@testing-library/react';
import Feature, {Feature_Title, Feature_SubTitle} from '../../components/Feature';

describe('<Feature />', () => {
    it('renders <Feature with populated data>', () => {
        const {container, getByText} = render(
        <Feature>
            <Feature_Title>Unlimited films, Tv Programmes and more.</Feature_Title>
            <Feature_SubTitle>Watch anywhere. Cancel at any time</Feature_SubTitle>
        </Feature>
        );

        expect(getByText('Unlimited films, Tv Programmes and more.')).toBeTruthy();
        expect(getByText('Watch anywhere. Cancel at any time')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });


    it('renders <Feature with just the title', () => {
        const {container, getByText, queryByText} = render(
        <Feature>
            <Feature_Title>Unlimited films, Tv Programmes and more.</Feature_Title>
        </Feature>
        );

        expect(getByText('Unlimited films, Tv Programmes and more.')).toBeTruthy();
        expect(queryByText('Watch anywhere. Cancel at any time')).toBeFalsy();
        expect(container.firstChild).toMatchSnapshot();
    });
});
