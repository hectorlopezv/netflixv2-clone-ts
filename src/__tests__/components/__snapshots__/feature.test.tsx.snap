// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders <Feature with just the title 1`] = `
<div
  class="Feature__Container-mtzwb2-0 gJHkQF"
>
  <h1
    class="Feature__Title-mtzwb2-1 hnDXvO"
  >
    Unlimited films, Tv Programmes and more.
  </h1>
</div>
`;

exports[`<Feature /> renders <Feature with populated data> 1`] = `
<div
  class="Feature__Container-mtzwb2-0 gJHkQF"
>
  <h1
    class="Feature__Title-mtzwb2-1 hnDXvO"
  >
    Unlimited films, Tv Programmes and more.
  </h1>
  <h2
    class="Feature__SubTitle-mtzwb2-2 ecLJlq"
  >
    Watch anywhere. Cancel at any time
  </h2>
</div>
`;
